submissions/TaniaBondarenko/a-tiny-js-world-pre-oop/index.js

```diff
@@ -0,0 +1,91 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== creatureECTS DEFINITIONS ========
+// Define your creatureects here
+const cat = {
+  species: "cat",
+  name: "Toby",
+  gender: "male",
+  legs: 4,
+  hands: 0,
+  saying: "Meow! Feed me!",
+  friends: "no one",
+};
+
+const dog = {
+  species: "dog",
+  name: "Rex",
+  gender: "male",
+  legs: 4,
+  hands: 0,
+  saying: "Woof-woof!",
+  friends: "",
+};
+
+const male = {
+  species: "human",
+  name: "Bill",
+  gender: "male",
+  legs: 2,
+  hands: 2,
+  saying: "Hi!",
+  friends: "",
+};
+
+const female = {
+  species: "human",
+  name: "Beverly",
+  gender: "female",
+  legs: 2,
+  hands: 2,
+  saying: "What a nice day!",
+  friends: "",
+};
+
+const catWoman = {
+  species: "cat-Woman",
+  name: "Maggie",
+  gender: "female",
+  legs: 2,
+  hands: 2,
+  friends: "",
+};
+
+catWoman.saying = cat.saying;
+
+dog.friends += `${female.name}, ${male.name}`;
+male.friends += `${female.name}, ${cat.name} - a ${cat.species}`;
+female.friends += `${dog.name} - a ${dog.species}`;
+catWoman.friends += `${cat.name} - a ${cat.species}`;
+
+const inhabitants = [cat, dog, male, female, catWoman];
+
+function printCreatureDatas(creature) {
+  print(["species", "name", "gender", "legs", "hands", "saying", "friends"].map((propertyName) => `${creature[propertyName]}`).join("; ") + `;`);
```

Comment:
Why `creature[propertyName]` needs to be wrapped in template literal?

Reply:
I need to use `join` and it works for `strings`..

Reply:
@TaniaBondarenko Why it needed to be so when it was `console.log`?

Reply:
@OleksiyRudenko , You are right. I believe it was consequences of my overthinking.

Reply:
@TaniaBondarenko i don't feel right or wrong. I was really interested. I try to find out where this pattern originates from. You would really help me if you told me. Will be much appreciated

Reply:
@OleksiyRudenko, I think it was somehow like that:
1. Firstly I've had literal string because it was necessary: `print(`${obj.species}; ${obj.name}; ${obj.gender}; ${obj.legs}; ${obj.hands}; ${obj.saying}; ${obj.friends}`);`
2. Then I replaced it with a short new version with  `map` and left literals with no paying attention to it: `${creature[propertyName]}`).join("; ") + `;
 It worked appropriately, with no errors, so as for me it couldn't be wrong.

Reply:
Thank you. It wasn't wrong, but rather redundant and note really helpful in reading code. This tiny things matter